src/test/prevector_tests.cpp

```diff
@@ -183,6 +183,22 @@ class prevector_tester {
         pre_vector = pre_vector_alt;
     }
 
+    void resize_uninitialized(realtype values) {
+        size_t s = real_vector.size() / 2;
+        real_vector.resize(s);
+        pre_vector.resize_uninitialized(s);
```

Comment:
Perhaps here you could make sure before going in loop to reserve values.size() many more memory to optimize the vector.

Reply:
@shahzadlone Thank you for your review.
Addressed that you pointed out.